### PR TITLE
add e2e test for etcd learner mode

### PR DESCRIPTION
**Pull request by pacoxu**
https://github.com/kubernetes/kubernetes/pull/115038 is pending on this.

- [ ] test-infra update in https://github.com/kubernetes/test-infra/pull/28574
/assign @neolit123

### CODE REVIEW

File: kinder/ci/tools/update-workflows/templates/workflows/learner-mode-tasks.yaml [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  - --copy-certs=auto
+  timeout: 10m
+- name: upgrade
+  description: |
+    upgrades the cluster to Kubernetes "upgradeVersion"
+  cmd: kinder
+  args:
+  - do
+  - kubeadm-upgrade
+  - --upgrade-version={{ .vars.kubernetesVersion }}
```
Comment (SataQiu) [inline kinder/ci/tools/update-workflows/templates/workflows/learner-mode-tasks.yaml:84]:
```suggestion
  - --upgrade-version={{ .vars.upgradeVersion }}
```

File: kinder/ci/workflows/learner-mode-tasks.yaml [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+  - --copy-certs=auto
+  timeout: 10m
+- name: upgrade
+  description: |
+    upgrades the cluster to Kubernetes "upgradeVersion"
+  cmd: kinder
+  args:
+  - do
+  - kubeadm-upgrade
+  - --upgrade-version={{ .vars.kubernetesVersion }}
```
Comment (SataQiu) [inline kinder/ci/workflows/learner-mode-tasks.yaml:85]:
```suggestion
  - --upgrade-version={{ .vars. upgradeVersion }}
```

File: kinder/ci/tools/update-workflows/templates/workflows/learner-mode.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+version: 1
+summary: |
+  This workflow tests the proper functioning of the {{ .KubernetesVersion }} version of both kubeadm and Kubernetes using
+  etcd learner mode to join control plane.
+  test grid > https://testgrid.k8s.io/sig-cluster-lifecycle-kubeadm#kubeadm-kinder-learner-mode{{ dashVer .KubernetesVersion }}
+  config    > https://git.k8s.io/test-infra/config/jobs/kubernetes/sig-cluster-lifecycle/{{ .TargetFile }}
+vars:
+  kubernetesVersion: "\{\{ resolve `ci/{{ ciLabelFor .KubernetesVersion }}` \}\}"
+  upgradeVersion: "\{\{ resolve `ci/{{ ciLabelFor .KubernetesVersion }}` \}\}"
```
Comment (SataQiu) [inline kinder/ci/tools/update-workflows/templates/workflows/learner-mode.yaml:9]: This means that we always upgrade to the current version, but that seems fine.

File: kinder/ci/tools/update-workflows/templates/workflows/learner-mode-tasks.yaml (hunk 1 above)
Comment (pacoxu) [reply kinder/ci/tools/update-workflows/templates/workflows/learner-mode-tasks.yaml]: Updated.

File: kinder/ci/tools/update-workflows/templates/workflows/learner-mode.yaml (hunk 3 above)
Comment (pacoxu) [reply kinder/ci/tools/update-workflows/templates/workflows/learner-mode.yaml:9]: The core part of this task is to join new control-planes. `upgrade` is just to make the process more complete.

**Review by SataQiu — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**